quickstarts/microsoft.compute/vm-simple-windows/main.bicep

```diff
@@ -116,18 +116,23 @@ resource vn 'Microsoft.Network/virtualNetworks@2021-02-01' = {
         addressPrefix
       ]
     }
+    subnets: [
+      {
+        name: subnetName
+        properties: {
+          addressPrefix: subnetPrefix
+          networkSecurityGroup: {
+            id: securityGroup.id
+          }
+        }
+      }
+    ]
   }
 }
 
-resource subnet 'Microsoft.Network/virtualNetworks/subnets@2021-02-01' = {
+resource subnet 'Microsoft.Network/virtualNetworks/subnets@2021-02-01' existing = {
```

Comment:
this is a really strange pattern - to create the subnets on the vnet then use an "existing" symbolic reference...  If we need to move the subnets to the vnet (to deal with idempotency issues) then we should just use the resourceId() function for the subnet id (or wait for the bicep changes to allow design-time property references).

Reply:
Thanks @bmoore-msft was interested to get your feedback on this.

FYI, these are the patterns that have come up so far talking over the options... or are used at some places in the current quickstarts.

https://github.com/Azure/bicep/issues/4653#issuecomment-930542087

Reply:
okay 7 out of the 8 PR's used that, they are now all updated.